builtin/providers/aws/resource_aws_elasticache_cluster.go

```diff
@@ -186,16 +186,18 @@ func resourceAwsElasticacheClusterCreate(d *schema.ResourceData, meta interface{
 		log.Printf("[DEBUG] Restoring Redis cluster from S3 snapshot: %#v", s)
 	}
 
-	_, err := conn.CreateCacheCluster(req)
+	resp, err := conn.CreateCacheCluster(req)
 	if err != nil {
 		return fmt.Errorf("Error creating Elasticache: %s", err)
 	}
 
+	d.SetId(*resp.CacheCluster.CacheClusterID)
+
 	pending := []string{"creating"}
 	stateConf := &resource.StateChangeConf{
 		Pending:    pending,
 		Target:     "available",
-		Refresh:    cacheClusterStateRefreshFunc(conn, d.Id(), "available", pending),
+		Refresh:    cacheClusterStateRefreshFunc(conn, *resp.CacheCluster.CacheClusterID, "available", pending),
```

Comment:
Can leave this as `d.Id()` since you moved the `SetId` up, but not a huge deal either way